py_trees/composites.py

```diff
@@ -549,78 +550,149 @@ class Parallel(Composite):
+            new_status = common.Status.FAILURE
         else:
-            if self.policy == common.ParallelPolicy.SUCCESS_ON_ALL:
-                if all([c.status == Status.SUCCESS for c in self.children]):
-                    new_status = Status.SUCCESS
-            elif self.policy == common.ParallelPolicy.SUCCESS_ON_ONE:
-                if any([c.status == Status.SUCCESS for c in self.children]):
-                    new_status = Status.SUCCESS
-        # special case composite - this parallel may have children that are still running
-        # so if the parallel itself has reached a final status, then these running children
-        # need to be made aware of it too
-        if new_status != Status.RUNNING:
+            if type(self.policy) is common.ParallelPolicy.SuccessOnAll:
+                if all([c.status == common.Status.SUCCESS for c in self.children]):
+                    new_status = common.Status.SUCCESS
+            elif type(self.policy) is common.ParallelPolicy.SuccessOnOne:
+                if any([c.status == common.Status.SUCCESS for c in self.children]):
+                    new_status = common.Status.SUCCESS
+            elif type(self.policy) is common.ParallelPolicy.SuccessOnSelected:
+                if all([c.status == common.Status.SUCCESS for c in self.policy.children]):
+                    new_status = common.Status.SUCCESS
+        # this parallel may have children that are still running
+        # so if the parallel itself has reached a final status, then
+        # these running children need to be terminated so they don't dangle
+        if new_status != common.Status.RUNNING:
             for child in self.children:
-                if child.status == Status.RUNNING:
+                if child.status == common.Status.RUNNING:
                     # interrupt it (exactly as if it was interrupted by a higher priority)
-                    child.stop(Status.INVALID)
+                    child.stop(common.Status.INVALID)
             self.stop(new_status)
         self.status = new_status
         yield self
 
     @property
     def current_child(self):
         """
-        Have to check if there's anything actually running first.
+        In some cases it's clear what the current child is, in others, there
+        is an ambiguity as multipe could exist. If the latter is true, it
+        will return the child relevant farthest down the list.
 
         Returns:
             :class:`~py_trees.behaviour.Behaviour`: the child that is currently running, or None
         """
-        if self.status == Status.INVALID:
+        if self.status == common.Status.INVALID:
             return None
-        if self.status == Status.FAILURE:
+        if self.status == common.Status.FAILURE:
             for child in self.children:
-                if child.status == Status.FAILURE:
+                if child.status == common.Status.FAILURE:
                     return child
             # shouldn't get here
```

Comment:
BTW this is obvious.

Reply:
Style thing ... I've been slowly converting the entire library to adhere somewhere more closely to the Python Zen Rule: _"Explicit is better than implicit"_.

Like the spelling fixes, no point in wasting time doing this all at once, but when I work on a new area, I continually evolve this and a few other things to where I want the library to be.